api/types/okta.go

```diff
@@ -393,6 +410,32 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 	o.Spec.CleanupTime = o.Spec.CleanupTime.UTC()
 	o.Spec.LastTransition = o.Spec.LastTransition.UTC()
 
+	// Deduplicate targets
+	allTargets := map[OktaAssignmentTargetV1_OktaAssignmentTargetType]map[string]struct{}{}
+
+	for _, target := range o.Spec.Targets {
+		if _, ok := allTargets[target.Type]; !ok {
+			allTargets[target.Type] = map[string]struct{}{}
+		}
+
+		allTargets[target.Type][target.Id] = struct{}{}
+	}
+
+	var deduplicatedTargets oktaAssignmentTargetsV1
+	for targetType, targets := range allTargets {
+		for target := range targets {
+			deduplicatedTargets = append(deduplicatedTargets, &OktaAssignmentTargetV1{
+				Type: targetType,
+				Id:   target,
+			})
+		}
+	}
+
+	// Sort the targets to provide some level of idempotency, i.e. two Okta assignments
+	// with the same metadata and same targets will be equal.
+	sort.Sort(deduplicatedTargets)
```

Comment:
You may find `slices.SortFunc` more ergonomic - you only need to provide the comparison and don't need to worry about implementing `sort.Interface` or defining a new type.